src/dataflow/components/nodes/factories/relay-rete-node-factory.tsx

```diff
@@ -27,18 +27,18 @@ export class RelayReteNodeFactory extends DataflowReteNodeFactory {
 
   public worker(node: NodeData, inputs: any, outputs: any) {
     const n1 = inputs.num1.length ? inputs.num1[0] : node.data.num1;
-    const result = +(n1 !== 0);   // convert all non-zero to 1
-
+    const recentValues: any = node.data.recentValues;
+    const mostRecentValue = recentValues && recentValues[recentValues.length - 1];
+    const lastValue = (mostRecentValue && mostRecentValue.nodeValue) ? mostRecentValue.nodeValue.val : 1;
+    // if there is not a valid input, use last value
+    // otherwise convert all non-zero to 1
+    const result = isNaN(n1) ? lastValue : +(n1 !== 0);
```

Comment:
`!isFinite()` would handle infinities as well, but it isn't as clearly a readability improvement because this code doesn't have the double-negative problem, so it's a judgment call.